### ISSUE TITLE
Support HTTPS sources

### ISSUE THREAD

**Issue by jsdw** [jsdw/weave#3]
Weave currently can proxy requests to https endpoints, but only supports serving up content on http at the moment.

Serving content on https would require either a pre existing or self signed certificate to be used.

While less useful perhaps during development, https is handy if you are demoing things you're hacking on and such, especially if any sort of sensitive data is sent either way. It would also be handy if weave added any sort of authentication (eg basic auth) support to restrict access to things being served up (use case: quick file server with basic access control)

**Comment by davidgraeff** [comment]
Hyper supports http2. Would support for Https connections (certificate management in weave) also mean http2 support? If so weave could be used for gRPC proxying, I guess.

**Comment by jsdw** [comment]
Interesting question; hyper is used to receive and send on http traffic and so I'd assume it at least supports http2, but it would need to be tested! If you do get the chance to try this, let me know what you find :)

**Comment by jsdw** [comment]
(I just wanted to note that weave can be used as a tcp proxy as well, which would transparently proxy any tcp traffic without caring about the contents, so this would allow basic http2 proxying, just without any routing bits)

**Comment by davidgraeff** [comment]
Absolutely. But the routing bits are the interesting parts. I'm still waiting for hyper to release a new version that uses stable futures-rs before I investigate any other code base.

**Comment by vi** [comment]
Is this going to be implemented?

What is maintainership status of `weave` in general?

**Comment by jsdw** [comment]
I don't have a use case for this support nowadays, but I'd happily accept a PR which adds it :)

I'm not actively adding features to weave at the moment (what it has is suitable for the sorts of things I use it for), but every now and then I may tweak and refine bits.
